src/components/v2/OnlinePreviewPanel.vue

```diff
@@ -0,0 +1,195 @@
+<script lang="ts" setup>
+import ThunderstoreMod from 'src/model/ThunderstoreMod';
+import { computed, ref, watch } from 'vue';
+import MarkdownRender from 'components/v2/MarkdownRender.vue';
+import { valueToReadableDate } from 'src/utils/DateUtils';
+import OnlineModList from 'components/views/OnlineModList.vue';
+import useStore from '../../store';
+import ThunderstoreDownloaderProvider, {
+    DependencySetBuilderMode
+} from 'src/providers/ror2/downloading/ThunderstoreDownloaderProvider';
+import ThunderstoreCombo from 'src/model/ThunderstoreCombo';
+import { getVersionAsThunderstoreVersion } from 'src/r2mm/manager/PackageDexieStore';
+import Link from 'components/Link.vue';
+
+const store = useStore();
+
+interface ModPreviewPanelProps {
+    mod: ThunderstoreMod;
+}
+
+const props = defineProps<ModPreviewPanelProps>();
+const readme = ref<string | null>(null);
+const changelog = ref<string | null>(null);
+const activeTab = ref<"README" | "CHANGELOG" | "Dependencies">("README");
+const loadingPanel = ref<boolean>(true);
+const dependencies = ref<ThunderstoreMod[]>([]);
+
+function setActiveTab(tab: "README" | "CHANGELOG" | "Dependencies") {
+    activeTab.value = tab;
+}
+
+function fetchDataFor(mod: ThunderstoreMod, type: "readme" | "changelog"): Promise<string> {
+    return fetch(`https://thunderstore.io/api/experimental/package/${mod.getOwner()}/${mod.getName()}/${mod.getLatestVersion()}/${type}/`)
+        .then(res => res.json())
+        .then(res => res.markdown);
+}
+
+function fetchReadme(modToLoad: ThunderstoreMod) {
+    return fetchDataFor(props.mod, "readme").then(res => {
+        if (props.mod === modToLoad) {
+            readme.value = res;
+        }
+    });
+}
+
+function fetchChangelog(modToLoad: ThunderstoreMod) {
+    return fetchDataFor(props.mod, "changelog").then(res => {
+        if (props.mod === modToLoad) {
+            changelog.value = res;
+        }
+    });
+}
+
+function fetchAll(modToLoad: ThunderstoreMod) {
+    setActiveTab("README");
+    loadingPanel.value = true;
+    buildDependencies(modToLoad)
```

Comment:
- Could consider error handlers in case of the Network errors
- Any specific reason for chaining the loading of the tab contents instead of triggering them all at once?
- If we want to keep them chained, should loading README be the first in the chain as it's always shown first?
- The way loadingPanel is used in the markup, all of the tabs show "fetching" until all of the tabs have finished loading (and get stuck to that if one of the other tabs fail to load)

Reply:
> Could consider error handlers in case of the Network errors

Valid. I can keep track and display relevant errors where appropriate in the form of the notification boxes if that approach sounds good?

> Any specific reason for chaining the loading of the tab contents instead of triggering them all at once?

It's just a lower number of requests in one hit. Node really doesn't like a large number of concurrent requests and these could pile up when combined with background downloads. I'm happy to change this though as it's only one extra call.

> If we want to keep them chained, should loading README be the first in the chain as it's always shown first?

Good spot, will change.

Reply:
> I can keep track and display relevant errors where appropriate in the form of the notification boxes if that approach sounds good?

Sounds good to me.